app/routes/session.js

```diff
@@ -61,6 +61,20 @@ function SessionHandler(db) {
             var invalidPasswordErrorMessage = "Invalid password";
             if (err) {
                 if (err.noSuchUser) {
+                    console.log('Error: attempt to login with invalid user: ', userName);
+
+                    // Fix for A1 - 3 Log Injection - encode/sanitize input for CSRF Injection
+                    // that could result in log forging:
+                    // - Step 1: Require a module that supports encoding
+                    // var ESAPI = require('node-esapi');
+                    // - Step 2: Encode the user input that will be logged in the correct context
+                    // following are a few examples:
+                    // console.log('Error: attempt to login with invalid user: %s', ESAPI.encoder().encodeForHTML(userName));
+                    // console.log('Error: attempt to login with invalid user: %s', ESAPI.encoder().encodeForJavaScript(userName));
+                    // console.log('Error: attempt to login with invalid user: %s', ESAPI.encoder().encodeForURL(userName));
+                    // or if you know that this is a CSRF vulnerability you can target this specifically as follows:
```

Comment:
Similar to above comment, do you mean CRLF instead of CSRF here?

Reply:
Yep, as above, meant CRLF.